### PR TITLE
C# bytes optimization

### PR DESCRIPTION
**Pull request by vroad**
Optimize Bytes with Buffer.BlockCopy, BitConverter and unsafe code.

### CODE REVIEW

**Comment by nadako** [comment]
We gotta benchmark the buffer.blockcopy vs array.copy thing, because quick googling suggests that it can actually be slower for some reason.

Also I'm not sure about the unsafe stuff, since I'm pretty sure some environments just don't allow this, so we might want to hide it behind a conditional compilation define.

@waneck could you look at this at some time?

**Comment by vroad** [comment]
Could you paste a link to a page that saying ArrayCopy is faster than BlockCopy?

I found this stackoverflow question when I googled about it. http://stackoverflow.com/a/33865267
This user pastes a link to a site saying that ArrayCopy is, but seems like he is testing this in a wrong way. I don't think it makes huge difference like "1.8x". And he does not explain why it's faster. Probably he should have tested this by launching test program times. http://stackoverflow.com/a/31139690

**Comment by vroad** [comment]
I think it already has "#if unsafe". Isn't that thing you are talking about?

Also, it might be faster to write/read floating points with `fixed `statements when unsafe is enabled. That will allow you to read/write floats with fewer statements. I'm no sure how big the overhead of `fixed` statement is, though it's may not be that huge.

File: std/haxe/io/Bytes.hx [hunk 1]
```diff
@@ -238,6 +243,17 @@ class Bytes {
 		#elseif cpp
 		if( pos < 0 || pos + 4 > length ) throw Error.OutsideBounds;
 		return untyped __global__.__hxcpp_memory_get_float(b,pos);
+		#elseif cs
+		#if unsafe
+		var value:UInt =
+			b[0 + pos] << 0 |
```
Comment (eliasku) [inline std/haxe/io/Bytes.hx]: why not just *((float*)(b + pos)) ?

**Comment by Simn** [comment]
Are there any benchmarks for this? I'm quite wary of optimization PRs otherwise...

**Comment by Simn** [comment]
Closing due to lack of response.

**Pull request closed without merging**